frb_codegen/src/generator/dart/ty_delegate.rs

```diff
@@ -102,14 +123,36 @@ impl TypeDartGeneratorTrait for TypeDelegateGenerator<'_> {
     }
 
     fn structs(&self) -> String {
-        if let IrTypeDelegate::PrimitiveEnum { ir, .. } = &self.ir {
-            super::TypeEnumRefGenerator {
+        match &self.ir {
+            IrTypeDelegate::PrimitiveEnum { ir, .. } => super::TypeEnumRefGenerator {
                 ir: ir.clone(),
                 context: self.context.clone(),
             }
-            .structs()
-        } else {
-            "".into()
+            .structs(),
+            IrTypeDelegate::GeneralArray { length, .. }
+            | IrTypeDelegate::PrimitiveArray { length, .. } => {
+                format!(
+                    "class {0} {{
```

Comment:
Replacing `List<T>` and typed lists with a class is an API-breaking change; perhaps have the new class implement [DelegatingList](https://pub.dev/documentation/collection/latest/collection/DelegatingList-class.html) so list methods are visible on it?